source/common/common/token_bucket_impl.h

```diff
@@ -29,8 +32,10 @@ class TokenBucketImpl : public TokenBucket {
   const double max_tokens_;
   const double fill_rate_;
   double tokens_;
+  absl::optional<bool> reset_once_;
   MonotonicTime last_fill_;
   TimeSource& time_source_;
+  absl::Mutex mutex_;
```

Comment:
I made in-place edits since the new class would be just a copy of this class which itself isn't used in too many places. Apart from fault filter, it's used only by the xDS grpc rate-limit. But I can definitely create a new class if code duplication is not a huge concern.

Reply:
@tonya11en I have made the token bucket thread-safety optional. Let me know if this approach looks good to you or not and I'll create a separate PR for these changes.

Reply:
Created new PR for the token bucket changes: https://github.com/envoyproxy/envoy/pull/14827